pkg/controller/projectreference/projectreference_adapter.go

```diff
@@ -324,6 +380,32 @@ func (r *ReferenceAdapter) createCredentials() error {
 	return nil
 }
 
+func (r *ReferenceAdapter) deleteCredentials() error {
+	secret := types.NamespacedName{
+		Namespace: r.projectClaim.Spec.GCPCredentialSecret.Namespace,
+		Name:      r.projectClaim.Spec.GCPCredentialSecret.Name,
+	}
+
+	// Check if the Secret exists or it's already deleted
+	if !gcputil.SecretExists(r.kubeClient, secret.Name, secret.Namespace) {
```

Comment:
I'm confused, `!gcputil.SecretExists` suggests we enter this branch if the secret **does not** exist. But the content of the branch tell me you're trying to delete it there, so it should be executed when it **does** exist.